pkg/assets/store.go

```diff
@@ -300,6 +300,34 @@ func (s *Store) GetKey(ctx context.Context, namespace string, sel monitoringv1.S
 	}
 }
 
+// GetKVMap processes the given SecretOrConfigMap selector list and returns the referenced key-content map.
+func (s *Store) GetKVMap(ctx context.Context, namespace string, sels []monitoringv1.SecretOrConfigMap) (map[string][]byte, error) {
+	m := make(map[string][]byte)
+	for _, sel := range sels {
+		switch {
+		case sel.Secret != nil:
+			value, err := s.GetSecretKey(ctx, namespace, *sel.Secret)
+			if err != nil {
+				return nil, err
+			}
+			if _, ok := m[sel.Secret.Key]; ok {
+				return nil, errors.Errorf("duplicated secret or configmap key %s in templates", sel.Secret.Key)
+			}
+			m[sel.Secret.Key] = []byte(value)
```

Comment:
> this doesn't work if the same key is present across different resources, e.g.
> 
> ```
> templates:
> - secret:
>     name: foo
>     key: template1
> - secret:
>     name: bar
>     key: template2
> ```

I didn't want this work if the same key is present across different resources. Because I use key as the template path/name, is shouldn't be the same. 
```go
		// set templates
		for tplName := range additionalData {
			cfgBuilder.cfg.Templates = append(cfgBuilder.cfg.Templates, tplName)
		}
```